runtime/j9vm/jvm.c

```diff
@@ -3650,25 +3650,38 @@ JVM_LoadSystemLibrary(const char *libName)
  */
 
 void* JNICALL
-JVM_LoadLibrary(const char* libName)
+JVM_LoadLibrary(const char *libName)
 {
-	void* result = NULL;
-	J9NativeLibrary* nativeLibrary = NULL;
-	J9JavaVM* javaVM = (J9JavaVM*)BFUjavaVM;
-	J9InternalVMFunctions* vmFuncs = javaVM->internalVMFunctions;
-	J9VMThread* currentThread = vmFuncs->currentVMThread(javaVM);
+	void *result = NULL;
+	J9JavaVM *javaVM = (J9JavaVM*)BFUjavaVM;
 
 	Trc_SC_LoadLibrary_Entry(libName);
-	vmFuncs->internalEnterVMFromJNI(currentThread);
-	/* registerBootstrapLibrary can't have VM access */
-	vmFuncs->internalReleaseVMAccess(currentThread);
-	if (vmFuncs->registerBootstrapLibrary(currentThread, libName, &nativeLibrary, FALSE) == J9NATIVELIB_LOAD_OK) {
-		result = (void*)nativeLibrary->handle;
-	}
-	vmFuncs->internalAcquireVMAccess(currentThread);
-	vmFuncs->internalExitVMToJNI(currentThread);
+	if (NULL == javaVM->applicationClassLoader) {
+		J9NativeLibrary *nativeLibrary = NULL;
+		J9InternalVMFunctions *vmFuncs = javaVM->internalVMFunctions;
+		J9VMThread *currentThread = vmFuncs->currentVMThread(javaVM);
```

Comment:
This can return NULL for unattached threads - might want to assert it here.

Reply:
Sure, will add assert.

Reply:
This PR is updated & passed `sanity.functional` & `extended.functional` at `AIX/zLinux/Linux AMD`.

@gacholio could you please review?